### PR TITLE
LWM2M and Coap changes v2

### PR DESCRIPTION
**Pull request by cabelitos**
Changes since v1:

* sol_coap_server_register_resource return an int now

### CODE REVIEW

File: src/samples/coap/simple-server.c [hunk 1]
```diff
@@ -237,7 +237,7 @@ main(int argc, char *argv[])
         return -1;
     }
 
-    if (!sol_coap_server_register_resource(server, &light, NULL)) {
+    if (sol_coap_server_register_resource(server, &light, NULL) < 0) {
         SOL_WRN("Could not register resource for the light");
```
Comment (brunobottazzini) [inline src/samples/coap/simple-server.c]: I know this is not in the patch but wouldn't this be better ? 

s/Could not register resource for the light/Could not register light resource/g

**Comment by brunobottazzini** [comment]
LGTM :+1:

File: src/samples/coap/simple-server.c (hunk 1 above)
Comment (cabelitos) [inline src/samples/coap/simple-server.c]: I can change that. NP

**Comment by glima** [comment]
+1

**Comment by cabelitos** [comment]
Merging this one.

**Pull request closed without merging**